### PR TITLE
C string formatting

### PR DESCRIPTION
**Pull request by javathunderman**
Reviving #1263. 

IA Page: https://duck.co/ia/view/string_fmt_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by javathunderman** [comment]
To avoid merge conflicts, please merge this PR first before merging #2168

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C String Formatting Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/string_fmt_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/c-string-formatting.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
+{
+    "id": "c_string_formatting_cheat_sheet",
+    "name": "C String Formatting",
+    "description": "For C-Style Formatting",
+    "metadata": {
+	"sourceName": "digilife",
+	"url": "http://www.digilife.be/quickreferences/QRC/C%20Reference%20Card%20%28ANSI%29%202.2.pdf"
+    },
+
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c-string-formatting.json]: @javathunderman I think we'll want to use the `terminal` template (maybe `code`)

File: share/goodie/cheat_sheets/json/c-string-formatting.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    "name": "C String Formatting",
+    "description": "For C-Style Formatting",
+    "metadata": {
+	"sourceName": "digilife",
+	"url": "http://www.digilife.be/quickreferences/QRC/C%20Reference%20Card%20%28ANSI%29%202.2.pdf"
+    },
+    
+    "section_order":[
+	"General Formatting",
+	"specifier \\(required\\)",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/c-string-formatting.json]: @javathunderman let's drop the `\\(required\\)` it's not needed here and doesn't match the section title below.

I think we should Title Case the sections as well to follow suite from General Formatting

**Comment by moollaza** [comment]
> To avoid merge conflicts, please merge this PR first before merging #2168

@javathunderman it's probably safest to just reset this branch to Master and cherry-pick your single commit -- that way we don't have to worry about that deletion PR

**Comment by javathunderman** [comment]
I'm combining this w/ #2168 into one PR. (Hooray! Two in one! :smile:)

**Comment by javathunderman** [comment]
@moollaza Hopefully this doesn't break CI...

**Comment by moollaza** [comment]
> I'm combining this w/ #2168 into one PR. (Hooray! Two in one! :smile:)

@javathunderman whoops, that's actually not allowed. The platform ensure that you only add one IA per PR. Otherwise we run into problems

**Pull request closed without merging**